dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/MongoDBRyaDAO.java

```diff
@@ -146,15 +153,18 @@ public void destroy() throws RyaDAOException {
     public void add(final RyaStatement statement) throws RyaDAOException {
         // add it to the collection
         try {
-            coll.insert(storageStrategy.serialize(statement));
-            for(final RyaSecondaryIndexer index: secondaryIndexers) {
-                index.storeStatement(statement);
+            final boolean canAdd = DocumentVisibilityUtil.doesUserHaveDocumentAccess(auths, statement.getColumnVisibility());
```

Comment:
Added exceptions for add and delete if they don't have the required authorizations.